modules/dnn/test/test_model.cpp

```diff
@@ -389,6 +428,44 @@ TEST_P(Test_Model, Segmentation)
     testSegmentationModel(weights_file, config_file, inp, exp, norm, size, mean, scale, swapRB);
 }
 
+TEST_P(Test_Model, TextRecognition)
+{
+    std::string imgPath = _tf("text_rec_test.png");
+    std::string weightPath = _tf("onnx/models/crnn.onnx", false);
+    std::string seq = "welcome";
+
+    Size size{100, 32};
+    double scale = 1.0 / 127.5;
+    Scalar mean = Scalar(127.5);
+    std::string decodeType = "CTC-greedy";
+    std::vector<std::string> vocabulary = {"0","1","2","3","4","5","6","7","8","9",
+                                           "a","b","c","d","e","f","g","h","i","j","k","l","m","n","o","p","q","r","s","t","u","v","w","x","y","z"};
+
+    testTextRecognitionModel(weightPath, "", imgPath, seq, decodeType, vocabulary, size, mean, scale);
+}
+
+TEST_P(Test_Model, TextDetection)
+{
+    std::string imgPath = _tf("text_det_test.png");
+    std::string weightPath = _tf("onnx/models/DB_TD500_resnet50.onnx", false);
+
+    // GroundTruth
+    std::vector<std::vector<Point>> gt = {{Point(213, 150), Point(133, 165), Point(142, 191), Point(221, 177)},
+                                          {Point(318, 72), Point(116, 116), Point(134, 164), Point(337, 119)}};
+
+    Size size{736, 736};
+    double scale = 1.0 / 255.0;
+    Scalar mean = Scalar(122.67891434, 116.66876762, 104.00698793);
+
+    float binThresh = 0.3;
+    float polyThresh = 0.5;
+    uint maxCandidates = 200;
+    int outputType = 0;
+    double unclipRatio = 2.0;
+
+    testTextDetectionModel(weightPath, "", imgPath, gt, outputType, binThresh, polyThresh, maxCandidates, unclipRatio, size, mean, scale);
+}
```

Comment:
Please test existing EAST model as well. Does it fit current API?

Reply:
https://github.com/opencv/opencv/blob/b45273eccb01323f5615b8900fd35407b1148939/modules/dnn/test/test_tf_importer.cpp#L817

Reply:
The API can support EAST now, and there is a code testing for EAST in test_model.cpp.
`TEST_P(Test_Model, TextDetectionByEAST)`

I have applied my APIs to samples/dnn/text_detection.cpp and tested it successfully with the previous models.
But, I am not sure about how to change the file name, because the text_detection.cpp and scene_text_spotting.cpp do the same thing with different algorithms. Do you have any suggestions?